openedx/core/djangoapps/credit/tests/test_views.py

```diff
@@ -469,24 +469,48 @@ def test_oauth(self):
         response = self.client.get(self.path, **headers)
         self.assertEqual(response.status_code, 200)
 
-    def test_create(self):
-        """ Verify the endpoint supports creating new CreditCourse objects. """
-        course_key = CourseKey.from_string('a/b/c')
+    def assert_course_created(self, course_id, response):
+        """ Verify an API request created a new CreditCourse object. """
         enabled = True
         data = {
-            'course_key': unicode(course_key),
+            'course_key': unicode(course_id),
             'enabled': enabled
         }
 
-        response = self.client.post(self.path, data=json.dumps(data), content_type=JSON)
         self.assertEqual(response.status_code, 201)
 
         # Verify the API returns the serialized CreditCourse
         self.assertDictEqual(json.loads(response.content), data)
 
         # Verify the CreditCourse was actually created
+        course_key = CourseKey.from_string(course_id)
         self.assertTrue(CreditCourse.objects.filter(course_key=course_key, enabled=enabled).exists())
 
+    def test_create(self):
```

Comment:
We could; but, since `put_as_create` requires a different URL, we reach a point where the savings of consolidating the methods is outweighed by the confusion of understanding them.